### PR TITLE
BROKERS: Repository - DeleteRepositoryAsync

### LINKED ISSUE #60: BROKERS: Repository - DeleteRepositoryAsync

### PR DESCRIPTION
**Pull request by Lezune**
Closes #60

### CODE REVIEW

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Resolve conflict

**Comment by Lezune** [comment]
> Resolve conflict

@cjdutoit, Merge conflicts has been resolved. Main branch has been updated so additional conflicts might need to be resolved once the review of the changes have been approved.

File: GitFyle.Core.Api/Brokers/Storages/IStorageBroker.Repository.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using GitFyle.Core.Api.Models.Foundations.Repositories;
+using System.Threading.Tasks;
+
+namespace GitFyle.Core.Api.Brokers.Storages
+{
+    internal partial interface IStorageBroker
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Storages/IStorageBroker.Repository.cs:10]: This is the wrong file. I was expecting an addition on the existing `IStorageBroker.Repositories.cs`

**Comment by Lezune** [comment]
Sorry! Still learning Github. Did not know that there was a re-review button in the top corner. Hopefully I did not slow things down.

**Review by cjdutoit — APPROVED** [review]

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.Repositories.cs [hunk 2]
```diff
@@ -5,13 +5,17 @@
...
 using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
 
 namespace GitFyle.Core.Api.Brokers.Storages
 {
     internal partial class StorageBroker
     {
         public DbSet<Repository> Repositories { get; set; }
 
+        public async ValueTask<Repository> DeleteRepositoryAsync(Repository repository) =>
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Storages/StorageBroker.Repositories.cs:16]: Keep CRUD sequence.  Delete should go last

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**